shotover-proxy/src/transforms/internal_debug_transforms.rs

```diff
@@ -1,23 +1,63 @@
 use crate::error::ChainResponse;
-use crate::message::Messages;
+use crate::message::{Message, MessageDetails, Messages};
+use crate::protocols::RawFrame;
+#[cfg(test)]
+use crate::transforms::Transforms;
 use crate::transforms::{Transform, Wrapper};
 use anyhow::anyhow;
+#[cfg(test)]
+use anyhow::Result;
 use async_trait::async_trait;
 use rand_distr::Distribution;
 use rand_distr::Normal;
+use redis_protocol::resp2::prelude::Frame;
+#[cfg(test)]
+use serde::Deserialize;
 use tokio::time::Duration;
 
+#[cfg(test)]
+#[derive(Deserialize, Debug, Clone)]
+pub struct DebugReturnerTransformConfig {
+    ok: bool,
+    response: Option<String>,
+}
+
+#[cfg(test)]
+impl DebugReturnerTransformConfig {
+    pub async fn get_source(&self) -> Result<Transforms> {
+        Ok(Transforms::DebugReturnerTransform(DebugReturnerTransform {
+            message: None,
+            response: self.response.clone(),
+            ok: self.ok,
+        }))
+    }
+}
+
 #[derive(Debug, Clone)]
 pub struct DebugReturnerTransform {
-    pub message: Messages,
+    pub message: Option<Messages>,
+    pub response: Option<String>,
     pub ok: bool,
 }
 
 #[async_trait]
 impl Transform for DebugReturnerTransform {
     async fn transform<'a>(&'a mut self, _message_wrapper: Wrapper<'a>) -> ChainResponse {
+        if let Some(response) = self.response.clone() {
```

Comment:
DebugReturnerTransform will first check if there is a response, then check if there is a message.

We should error when creating if there is both a response and message.

Reply:
We should use enums so that is impossible to be in invalid state

Reply:
Fixed